src/hera/workflows/container.py

```diff
@@ -34,6 +35,9 @@ class Container(
     security_context: Optional[SecurityContext] = None
     working_dir: Optional[str] = None
 
+    def __hera_hooks__(self):
+        GlobalConfig.dispatch_hooks(self)
```

Comment:
@flaviuvadan we might have a circular import issue if we import global config here and import these objects back in global config. Can we break this dependency by introducing a protocol in a different third module and using that in appropriate places.

Reply:
I will add some tests. I do not think this is the case because we import the objects in the config file only when performing type checking

Reply:
Ah, no, I was wrong.. the circular import is a problem because we have `isinstance` calls